src/web/core2numpy.rkt

```diff
@@ -0,0 +1,111 @@
+#lang racket
+
+(require fpbench/src/imperative)
+
+(provide core->numpy)
+
+(define numpy-reserved   ; Language-specific reserved names (avoid name collision)
+  '(False None True and as assert break class continue def del
+    elif else except finally for from global if import in is
+    lambda nonlocal not or pass raise return try while with yield))
+
+(define (operator->numpy op args ctx)
+  (match (cons op args)
+   [(list 'deg2rad a) (format "numpy.deg2rad(~a)" a)]
+   [(list 'rad2deg a) (format "numpy.rad2deg(~a)" a)]
+   [(list 'logaddexp a b) (format "numpy.logaddexp(~a, ~a)" a b)]
+   [(list 'logaddexp2 a b) (format "numpy.logaddexp2(~a, ~a)" a b)]
+   [(list 'square a) (format "numpy.square(~a)" a)]
+   [(list 'asin a) (format "numpy.arcsin(~a)" a)]
+   [(list 'acos a) (format "numpy.arccos(~a)" a )]
+   [(list 'atan a) (format "numpy.arctan(~a)" a )]
+   [(list 'asinh a) (format "numpy.arcsinh(~a)" a )]
+   [(list 'acosh a) (format "numpy.arccosh(~a)" a )]
+   [(list 'atanh a) (format "numpy.arctanh(~a)" a )]
+   [(list 'atan2 a b) (format "numpy.arctan2(~a, ~a)" a b)]
+   [(list '+ a b) (format "numpy.add(~a, ~a)" a b)]
+   [(list 'recip a) (format "numpy.reciprocal(~a)" a )]
+   [(list 'neg a) (format "numpy.negative(~a)" a )]
+   [(list '* a b) (format "numpy.multiply(~a, ~a)" a b)]
+   [(list '/ a b) (format "numpy.divide(~a, ~a)" a b)]
+   [(list 'pow a b) (format "numpy.float_power(~a, ~a)" a b)]
+   [(list '- a b) (format "numpy.subtract(~a, ~a)" a b)]
+   [(list '- a) (format "numpy.negative(~a)" a)]
+   [(list 'rint a) (format "numpy.rint(~a)" a )] 
+   [(list '> a b) (format "numpy.greater(~a, ~a)" a b)]
+   [(list '>= a b) (format "numpy.greater_equal(~a, ~a)" a b)]
+   [(list '< a b) (format "numpy.less(~a, ~a)" a b)]
+   [(list '<= a b) (format "numpy.less_equal(~a, ~a)" a b)]
+   [(list '== a b) (format "numpy.equal(~a, ~a)" a b)]
+   [(list '!= a b) (format "numpy.not_equal(~a, ~a)" a b)]
+   [_ (format "numpy.~a(~a)" op (string-join args ", "))]))
+
+(define (constant->numpy x ctx)
+  (define prec (ctx-lookup-prop ctx ':precision))
+  (match x
+   ['TRUE "True"]
+   ['FALSE "False"]
+   ['INFINITY (format "numpy.ones(length, dtype=~a)*numpy.inf" (type->numpy prec))]
+   ['NAN (format "numpy.ones(length, dtype=~a)*numpy.nan" (type->numpy prec))]
+   ['PI (format "numpy.ones(length, dtype=~a)*numpy.pi" (type->numpy prec))]
+   ['E (format "numpy.ones(length, dtype=~a)*numpy.e" (type->numpy prec))]
+   [(? hex?) (format "numpy.ones(length, dtype=~a)*~a" (type->numpy prec) (real->double-flonum (hex->racket x)))]
+   [(? number?) (format "numpy.ones(length, dtype=~a)*~a" (type->numpy prec) (real->double-flonum x))]
+   [(? symbol?) (format "numpy.ones(length, dtype=~a)*~a" (type->numpy prec) x)]))
+
+(define declaration->numpy
```

Comment:
I'm not 100% sure we need to output anything at all for declarations. The declarations are intended for compiling C code like:

```
int t;
if (foo) {
  t = ...;
} else {
  t = ...;
}
```

In Python you don't have to declare at all, so maybe returning an empty string would work.

Reply:
The core2python3-10 compiler had a declaration->python which is why I kept this in

Reply:
Fine